pkg/build/generator/generator.go

```diff
@@ -575,25 +575,54 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from kapi
 	glog.V(4).Infof("Resolving ImageStreamReference %s of Kind %s in namespace %s", from.Name, from.Kind, namespace)
 	switch from.Kind {
 	case "ImageStreamImage":
-		imageStreamImage, err := g.Client.GetImageStreamImage(kapi.WithNamespace(ctx, namespace), from.Name)
+		name, id, err := imageapi.ParseImageStreamImageName(from.Name)
 		if err != nil {
 			err = resolveError(from.Kind, namespace, from.Name, err)
 			glog.V(2).Info(err)
 			return "", err
 		}
-		image := imageStreamImage.Image
-		glog.V(4).Infof("Resolved ImageStreamReference %s to image %s with reference %s in namespace %s", from.Name, image.Name, image.DockerImageReference, namespace)
-		return image.DockerImageReference, nil
+		stream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), name)
+		if err != nil {
+			err = resolveError(from.Kind, namespace, from.Name, err)
+			glog.V(2).Info(err)
+			return "", err
+		}
+		tag, _ := imageapi.LatestImageTagEvent(stream, id)
+		if len(tag) == 0 {
+			err = resolveError(from.Kind, namespace, from.Name, fmt.Errorf("unable to find tag associated with image %q", id))
+			glog.V(2).Info(err)
+			return "", err
+		}
+		reference, ok := imageapi.ResolveLatestTaggedImage(stream, tag)
```

Comment:
@miminar you mean to say that the "tag" argument to ResolveLatestTaggedImage is being effectively ignored?  That wasn't my interpretation of how ResolveLatestTaggedImage works, but if it is that sounds like bad behavior for the helper function and not aligned with the godoc which says:

```
ResolveLatestTaggedImage returns the appropriate pull spec for a given tag in
the image stream, handling the tag's reference policy if necessary to return
a resolved image.
```